gammapy/maps/wcs.py

```diff
@@ -459,6 +459,45 @@ def footprint(self):
         coords = self.wcs.calc_footprint()
         return SkyCoord(coords, frame=self.frame, unit="deg")
 
+    @classmethod
+    def from_aligned(cls, geom, skydir, width):
+        """Create an aligned geometry from an existing one
+
+        Parameters
+        ----------
+        geom : `~WcsGeom`
+            A reference WCS geometry object.
+        skydir : tuple or `~astropy.coordinates.SkyCoord`
+            Sky position of map center.  Can be either a SkyCoord
+            object or a tuple of longitude and latitude in deg in the
+            coordinate system of the map.
+        width : float or tuple or list or string
+            Width of the map in degrees.  A tuple will be interpreted
+            as parameters for longitude and latitude axes.  For maps
+            with non-spatial dimensions, list input can be used to
+            define a different map width in each image plane.
+
+        Returns
+        -------
+        geom : `~WcsGeom`
+            An aligned WCS geometry object with specified size and center.
+
+        """
+        width = _check_width(width) * u.deg
+        xref, yref = geom.to_image().coord_to_idx(skydir)
+        npix = np.rint(width / geom.pixel_scales).astype(int)
+        xref = -int(xref) + npix[0] // 2 + geom.wcs.wcs.crpix[0]
+        yref = -int(yref) + npix[1] // 2 + geom.wcs.wcs.crpix[1]
+        return cls.create(
```

Comment:
Does it really work like this? 
If one uses a "TAN" projection, the center will be changed. No? 
Is the WCS really the same as that obtained from a cutout for such a projection? 

The test works, but what happens if you take a significantly larger distance for the cutout geometry.

Reply:
Good point. Indeed this can only work up to certain numerical precision. I'll add an edge case as test as well.

Reply:
Why not just rely on `cutout` here? It guarantees to have an aligned WCS.

Reply:
Because `.cutout()` will clip at the data edges? But in general you are right, we should do the same operation as in `Cutout2D`. I'll take a look at the implementation and see how I can improve. I think the current implementation (also because of the spurious `-1` is not yet correct...)

Reply:
I managed to find the consistent implementation with `Cutout2D` and added a test case using a TAN projection and a reference point further away. I think this should be fine for now.